packages/dcos-history/extra/history/server.py

```diff
@@ -1,93 +1,8 @@
-import logging
 import os
-import sys
+import history.server_util
```

Comment:
The goal long term would be to move to aiohttp and coroutines for the helper rather than mixing threading and python. The old threading is a big hack as well (and that it worked is some very specific trickery). 

All the refactoring and code shape changes this gets get us closer to what that will need to be (see https://github.com/dcos/dcos/blob/master/dcos_installer/async_server.py)

Gunicorn is how we run the other servers we have. That flask doesn't bail on port bind failing is a bug in flask. Can't really fix that.

Doing a "is this port available" check and then doing the "start flask" doesn't actually fix the issue. Flask's loop may still start, someone else grabbed the port between the "check port" and "start flask", flask continues in a loop while not actually listening anywhere.